scripts/inline_comments_data_collection.py

```diff
@@ -136,18 +107,25 @@ def process_comments(limit, diff_length_limit):
             if not most_recent_update:
                 continue
 
-            fix_patch_id, revision_phid = get_diff_info_from_phid(
-                most_recent_update["fields"]["new"]
-            )
+            fix_patch_id = diff_phid_to_revisions_map.get(
+                most_recent_update["fields"].get("new"), [None]
```

Comment:
In which case could we have no diff ID?

Reply:
Given we're getting `most_recent_update` from the revisions that are in the DB, shouldn't we always find it?

Reply:
That would make sense, but the mapping does not contain all diff ids/phids to revisions and I encounter `KeyError` when running it locally if I do not have the `.get()`. Will look into this.

Reply:
Try finding a specific example of such a case and then it'll be easier to investigate and find why it's missing

Reply:
I think it's because in the transaction of type `udpate`, it contains the `diffPHID` for the old and new patches, but not the `diffID`. So unless there is a dataset that contains all diffs with their `diffPHID` and `diffID`, then I'm not sure the map will contain all `diffID` for all `diffPHID` (i.e. a mapping/conversion between `diffPHID` and `diffID`.

I've tried populating the map with the `diffPHID` and `diffID` of the revision itself (`revision["fields"]["diffPHID"]` and `revision["fields"]["diffID"]`) but still encountered the same `KeyError`.